### ISSUE TITLE
Remote cluster client role is not documented

### ISSUE THREAD

**Issue by fdartayre** [elastic/elasticsearch#62445]
**Elasticsearch version** (`bin/elasticsearch --version`): 7.9.1

**Description of the problem including expected versus actual behavior**:
https://www.elastic.co/guide/en/elasticsearch/reference/7.9/modules-node.html doesn't mention the `remote_cluster_client` role.

In [node roles](https://www.elastic.co/guide/en/elasticsearch/reference/7.9/modules-node.html#node-roles), the documentaion also states:
>You can define the roles of a node by setting `node.roles`. If you don’t configure this setting, then the node has the following roles by default: `master`, `data`, `ingest`, `ml`.

Which is wrong, since nodes are also configured with `remote_cluster_client`  by default.

**Comment by jrodewig** [comment]
Hi @fdartayre 

We cover `node.remote_cluster_client` in our remote cluster settings here:
https://www.elastic.co/guide/en/elasticsearch/reference/7.9/modules-remote-clusters.html#remote-cluster-settings

That content doesn't seem like a great fit under node roles as it's not an actual node role, but a separate node setting.

@lockewritesdocs -- You recently touched these. Any additional thoughts on this? If not, I can close this ticket.

**Comment by lockewritesdocs** [comment]
Hmm, this is interesting. We indicate that:
>By default, any node in the cluster can act as a cross-cluster client and connect to remote clusters.

Users can toggle whether a cluster can be a remote cluster by setting `node.remote_cluster_client` to `false`, but I think that whether a node is eligible to be a remote cluster is different than their node _role_. I agree with James that this setting is separate. 

@fdartayre, when might users set `node.remote_cluster_client` to `false`? I'm just trying to understand the connection between setting node roles and configuring remote cluster eligibility.

**Comment by fdartayre** [comment]
@lockewritesdocs from my experience, explicitly disabling remote cluster client (setting `node.remote_cluster_client` to `false`) was not very common. 

The main problem I see with not mentioning `remote_cluster_client` in https://www.elastic.co/guide/en/elasticsearch/reference/7.9/modules-node.html is the following. When setting node roles via `node.roles: [ ]` instead of `node.<role>: true|false`, if one configures something like this:
```
node.roles: [ master, data, ingest ]
```
It will disable remote cluster connections. So the user won't be able to configure remote clusters (they have disabled the default setting by not explicitly configuring it), which is probably not what they expected to do.

**Comment by jrodewig** [comment]
@lockewritesdocs I did some testing this morning and @fdartayre is right. It looks like `remote_cluster_client` is a node role although it's not been documented as such.

Here's how I tested it:

1. Start up a 7.x cluster and run a nodes info API request.

```
GET /_nodes/settings
``` 

The response will list the default node roles. Note `remote_cluster_client` is included in the list.

```
...
      "roles" : [
        "data",
        "ingest",
        "master",
        "ml",
        "remote_cluster_client",
        "transform"
      ],
...
```
2. Shut down the cluster and add this to `elasticsearch.yml`:

```
node.roles: [ master, data, ingest ]
```

3.  Restart the cluster and rerun the nodes info API request.
```
GET /_nodes/settings
``` 

The response will no longer list `remote_cluster_client` in the node roles list.
```
...
      "roles" : [
        "master",
        "data",
        "ingest"
      ],
...
```

Seems like pretty trappy behavior, and we should document `remote_cluster_client` as a role. Let me know if you want me to take care of the update. Thanks for raising @fdartayre.
